chore(docs): Update "Upgrading NodeJS" (#30128)
gatsbyjs/gatsby: docs/docs/upgrading-node-js.md

```shell
node -v
v12.21.0
```

This example shows Node.js version 12, specifically v12.21.0.

## Upgrading from Node.js version 10

> Node.js version 10 reached _End-of-life_ status on April 30, 2021. Many of Gatsby's dependencies are updating to Node.js version 12 and above. Gatsby must also update in order to deliver new features and bug fixes more quickly.

Reviewer: Node 10 didn't reach end of life yet (it still have almost 2 months left)

Author: I was pragmatic here. Rather than writing `reaches` and forgetting about it, I kept it `reached` and it's slightly incorrect for 2 months. But we can rephrase it:

```suggestion
Node.js version 10 _End-of-life_ date is April 30, 2021. Many of Gatsby's dependencies are updating to Node.js version 12 and above. Gatsby must also update in order to deliver new features and bug fixes more quickly.
```

Author: @pieh done 👍

Gatsby also relies on a huge ecosystem of JavaScript dependencies. As the ecosystem moves away from older, unsupported Node.js versions we have to keep pace to ensure that bugs can be fixed and new features can be released.

> In this document, you learned how you upgrade from Node.js version 10 (which has reached _End of Life_ status) to Node.js version 12.

Reviewer: same as comment about
> **reached** _End of Life_ status`
